plugin-health-scoring/Jenkinsfile

```diff
@@ -0,0 +1,59 @@
+def cronExpr = env.BRANCH_IS_PRIMARY ? '@daily' : ''
+def reportsFolder = 'plugin-health-scoring'
+def etagsFile = 'etags.txt'
+def reportFile = env.BRANCH_IS_PRIMARY ? 'scores.json' : "scores-${env.BRANCH_NAME}.json"
+def reportLines = 0
+
+pipeline {
+  agent {
+    label 'jnlp-linux-arm64'
+  }
+
+  options {
+    buildDiscarder logRotator(daysToKeepStr: '90')
+    lock(resource: 'plugin-health-scoring', inversePrecedence: true)
+    timeout(time: 5, unit: 'MINUTES')
+    disableConcurrentBuilds()
+  }
+
+  triggers {
+    cron( cronExpr )
+  }
+
+  stages {
+    stage('Fetch API') {
+      environment {
+        REPORTS_FOLDER = "${reportsFolder}"
+        ETAGS_FILE = "${etagsFile}"
+        REPORT_FILE = "${reportFile}"
+        HEALTH_URL = 'https://plugin-health.jenkins.io/api/scores'
+      }
+
+      steps {
+        dir('plugin-health-scoring') {
+          sh 'bash fetch-report.sh'
+          // Retrieve the amount of lines in the report file to conditionally publish
+          script {
+            reportLines = sh(returnStdout:true, script: '''
+            wc -l "${REPORT_FILE}" | awk '{print $1}'
+            ''').trim()
```

Comment:
Suggestion: could fetch-report.sh returns the amount of lines so we could use a single sh script block?

Reply:
That would make the system way more fragile: we would need to specifically design the script to remove all standard output (or send it to standard error) which not only is a waste of time but also hard to maintain.

With the "2 commands" pattern, we apply POSIX principles: 1 step is to generate the report, the other is specific to Jenkins pipeline to retrieve an information (for conditional publication).